File: BGC_WebApi/BGC_WebApi/DTOs/GameSessionRegistrationDTO.cs
using BGC_DataAccess.Entities;

namespace BGC_WebApi.DTOs;
/// <summary>
/// DTO for game session registration
/// </summary>
public class GameSessionRegistrationDTO
{   
    /// <summary>
    /// Id
    /// </summary>
    public int GameSessionRegistrationId { get; set; }
    /// <summary>
    /// Id of the player
    /// </summary>
    public int MemberId { get; set; }
    /// <summary>
    /// Player
    /// </summary>
    public Member Player { get; set; }
    /// <summary>
    /// Id of the game session
    /// </summary>
    public int GameSessionId { get; set; }
    /// <summary>
    /// Game session
    /// </summary>
    public GameSession GameSession { get; set; }
}

File: BGC_Portal/src/app/api/models/game-session-registration-dto.ts
/* tslint:disable */
/* eslint-disable */
import { GameSessionDto } from './game-session-dto';
import { MemberDto } from './member-dto';
export interface GameSessionRegistrationDto {
  gameSession?: GameSessionDto;
  gameSessionId?: number;
  gameSessionRegistrationId?: number;
  memberId?: number;
  player?: MemberDto;
}
